### ISSUE TITLE
Intel Skylake is not autodetected.

### ISSUE THREAD

**Issue by devinamatthews** [flame/blis#55]
On my i5-6500, the configuration is auto-detected as 'reference'. Since there is no skylake ukernel, the config should be set to haswell.

**Comment by jeffhammond** [comment]
How does autodetect work (yes, I am too lazy to look at the code)?  Shouldn't it just look at `/proc/cpuinfo` by default?  Frankly, I do not think using Intel architecture code names for ukernels is the best method.  For reference, the Intel compiler describes platforms by ISA as shown below.  I wonder if it might be better to use AVX, AVX2, MIC-AVX512 (shorthand KNL if necessary), etc. for ukernels.

```
-ax<code1>[,<code2>,...]
          generate code specialized for processors specified by <codes>
          while also generating generic IA-32 instructions.  
          <codes> includes one or more of the following:
            SSSE3   May generate Intel(R) SSSE3, SSE3, SSE2, and SSE
                    instructions for Intel processors.
            SSE4.1  May generate Intel(R) SSE4.1, SSSE3, SSE3, SSE2, and SSE
                   instructions for Intel processors.
            SSE4.2  May generate Intel(R) SSE4.2, SSE4.1, SSSE3, SSE3, SSE2,
                    and SSE instructions for Intel processors.
            AVX     May generate Intel(R) Advanced Vector Extensions (Intel(R)
                    AVX), Intel(R) SSE4.2, SSE4.1, SSSE3, SSE3,
                    SSE2, and SSE instructions for Intel(R) processors.
            CORE-AVX2
                    May generate Intel(R) Advanced Vector Extensions 2
                    (Intel(R) AVX2), Intel(R) AVX, SSE4.2, SSE4.1, SSSE3, SSE3,
                    SSE2, and SSE instructions for Intel(R) processors.
            CORE-AVX-I
                    May generate Intel(R) Advanced Vector Extensions (Intel(R)
                    AVX), including instructions in Intel(R) Core 2(TM)
                    processors in process technology smaller than 32nm,
                    Intel(R) SSE4.2, SSE4.1, SSSE3, SSE3, SSE2, and SSE
                    instructions for Intel(R) processors.
            MIC-AVX512
                    May generate Intel(R) Advanced Vector Extensions 512
                    (Intel(R) AVX-512) Foundation instructions, Intel(R)
                    AVX-512 Conflict Detection instructions, Intel(R) AVX-512
                    Exponential and Reciprocal instructions, Intel(R) AVX-512
                    Prefetch instructions for Intel(R) processors, and the
                    instructions enabled with CORE-AVX2.
            CORE-AVX512 
                    May generate Intel(R) Advanced Vector Extensions 512 
                    (Intel(R) AVX-512) Foundation instructions, Intel(R) 
                    AVX-512 Conflict Detection instructions, Intel(R) AVX-512
                    Doubleword and Quadword instructions, Intel(R) AVX-512 
                    Byte and Word instructions and Intel(R) AVX-512 Vector 
                    Length Extensions for Intel(R) processors, and the 
                    instructions enabled with CORE-AVX2. 
```

**Comment by devinamatthews** [comment]
I think on x86 it looks at cpuid, so the family bits have to be checked 
explicitly. This should(?) be an easy fix for whomever is familiar with 
that code. As for naming configs, as it is now the config determines 
both the ukernel AND the cache blocking parameters. Thus, more than one 
config may be needed even for the same ISA (e.g. Carrizo and 
Piledriver). BTW does desktop Skylake (which does not have AVX512) 
nevertheless get the full 32 ymm registers?

**Comment by jeffhammond** [comment]
`cpuid` is at least as useful as `/proc/cpuinfo` (which is based upon `cpuid` AFAIK), so that should be fine.

It would be nice to separate ukernel and cache parameters in the config detection, because e.g. your Skylake i5-6500 and my Haswell E5-2699v3 server are going to use the same AVX2 ukernel but have very different cache configurations.

**Comment by xianyi** [comment]
@devinamatthews , I will submit a patch for auto detecting skylake.

**Comment by devinamatthews** [comment]
Fixed by #108.
